agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiComponentInstaller.java

```diff
@@ -228,8 +228,15 @@ private static AppIdSupplier start(Instrumentation instrumentation) {
       RpConfigurationPolling.startPolling(rpConfiguration, config, telemetryClient, appIdSupplier);
     }
 
-    // initialize StatsbeatModule
-    statsbeatModule.start(telemetryClient, config);
+    // initialize StatsbeatModule and don't start Statsbeat when proxy is used
+    String configProxyHost = config.proxy.host;
+    String systemPropertyProxyHost = System.getProperty("https.proxyHost");
+    if (configProxyHost == null
+        || configProxyHost.isEmpty()
+        || systemPropertyProxyHost == null
+        || systemPropertyProxyHost.isEmpty()) {
+      statsbeatModule.start(telemetryClient, config);
+    }
```

Comment:
with no-logging and kill-switch after failures, I don't think we need this anymore?

Reply:
well, errorhandler is for exception from breeze, errorcode is handled differently. that IcM is returning 403 for statsbeat because of private points not AMPL. Error handler is for AMPL case. we still need to handle proxy case though.

Reply:
ah, I was assuming that we would kill-switch on three consecutive "failures" meaning anything other than 200?